### ISSUE TITLE
Android Builds Fails to Run Properly

### ISSUE THREAD

**Issue by OmarAYousry** [microsoft/MixedReality-WebRTC#441]
**Describe the bug**
_When I launch the VideoChatDemo on Android, I can't see/hear myself, or connect to any peer_

**Expected behavior**
_To be able to see myself, then be able to connect to the remote peer on the Unity Editor_

**Error Logs**
_I got these messages by running android sdk's monitor.bat and filtering for Unity messages_
<details open><summary>On Boot, normal messages (i.e. Java Unity found, followed by the errors)</summary>
<code>
07-06 11:04:51.249: E/Unity(13663): AndroidJavaException: java.lang.UnsatisfiedLinkError: No implementation found for void org.webrtc.PeerConnectionFactory.nativeInitializeAndroidGlobals() (tried Java_org_webrtc_PeerConnectionFactory_nativeInitializeAndroidGlobals and Java_org_webrtc_PeerConnectionFactory_nativeInitializeAndroidGlobals__)
07-06 11:04:51.249: E/Unity(13663): java.lang.UnsatisfiedLinkError: No implementation found for void org.webrtc.PeerConnectionFactory.nativeInitializeAndroidGlobals() (tried Java_org_webrtc_PeerConnectionFactory_nativeInitializeAndroidGlobals and Java_org_webrtc_PeerConnectionFactory_nativeInitializeAndroidGlobals__)
07-06 11:04:51.249: E/Unity(13663): 	at org.webrtc.PeerConnectionFactory.nativeInitializeAndroidGlobals(Native Method)
07-06 11:04:51.249: E/Unity(13663): 	at org.webrtc.PeerConnectionFactory.initialize(PeerConnectionFactory.java:217)
07-06 11:04:51.249: E/Unity(13663): 	at com.unity3d.player.UnityPlayer.nativeRender(Native Method)
07-06 11:04:51.249: E/Unity(13663): 	at com.unity3d.player.UnityPlayer.c(Unknown Source:0)
07-06 11:04:51.249: E/Unity(13663): 	at com.unity3d.player.UnityPlayer$e$2.queueIdle(Unknown Source:72)
07-06 11:04:51.249: E/Unity(13663): 	at android.os.MessageQueue.next(MessageQueue.java:405)
07-06 11:04:51.249: E/Unity(13663): 	at android.os.Looper.loop(Loo
07-06 11:04:51.269: I/Unity(13663): NodeDSS local peer ID : UNITY
07-06 11:04:51.269: I/Unity(13663):  
07-06 11:04:51.269: I/Unity(13663): (Filename: ./Runtime/Export/Debug.bindings.h Line: 45)
07-06 11:04:51.292: I/Unity(13663): NodeDSS remote peer ID : PC
07-06 11:04:51.292: I/Unity(13663):  
07-06 11:04:51.292: I/Unity(13663): (Filename: ./Runtime/Export/Debug.bindings.h Line: 45)
07-06 11:04:51.322: E/Unity(13663): AndroidJavaException: java.lang.UnsatisfiedLinkError: No implementation found for void org.webrtc.PeerConnectionFactory.nativeInitializeAndroidGlobals() (tried Java_org_webrtc_PeerConnectionFactory_nativeInitializeAndroidGlobals and Java_org_webrtc_PeerConnectionFactory_nativeInitializeAndroidGlobals__)
07-06 11:04:51.322: E/Unity(13663): java.lang.UnsatisfiedLinkError: No implementation found for void org.webrtc.PeerConnectionFactory.nativeInitializeAndroidGlobals() (tried Java_org_webrtc_PeerConnectionFactory_nativeInitializeAndroidGlobals and Java_org_webrtc_PeerConnectionFactory_nativeInitializeAndroidGlobals__)
07-06 11:04:51.322: E/Unity(13663): 	at org.webrtc.PeerConnectionFactory.nativeInitializeAndroidGlobals(Native Method)
07-06 11:04:51.322: E/Unity(13663): 	at org.webrtc.PeerConnectionFactory.initialize(PeerConnectionFactory.java:217)
07-06 11:04:51.322: E/Unity(13663): 	at com.unity3d.player.UnityPlayer.nativeRender(Native Method)
07-06 11:04:51.322: E/Unity(13663): 	at com.unity3d.player.UnityPlayer.c(Unknown Source:0)
07-06 11:04:51.322: E/Unity(13663): 	at com.unity3d.player.UnityPlayer$e$2.queueIdle(Unknown Source:72)
07-06 11:04:51.322: E/Unity(13663): 	at android.os.MessageQueue.next(MessageQueue.java:405)
07-06 11:04:51.322: E/Unity(13663): 	at android.os.Looper.loop(Loo
07-06 11:04:51.324: E/Unity(13663): AndroidJavaException: java.lang.UnsatisfiedLinkError: No implementation found for void org.webrtc.PeerConnectionFactory.nativeInitializeAndroidGlobals() (tried Java_org_webrtc_PeerConnectionFactory_nativeInitializeAndroidGlobals and Java_org_webrtc_PeerConnectionFactory_nativeInitializeAndroidGlobals__)
07-06 11:04:51.324: E/Unity(13663): java.lang.UnsatisfiedLinkError: No implementation found for void org.webrtc.PeerConnectionFactory.nativeInitializeAndroidGlobals() (tried Java_org_webrtc_PeerConnectionFactory_nativeInitializeAndroidGlobals and Java_org_webrtc_PeerConnectionFactory_nativeInitializeAndroidGlobals__)
07-06 11:04:51.324: E/Unity(13663): 	at org.webrtc.PeerConnectionFactory.nativeInitializeAndroidGlobals(Native Method)
07-06 11:04:51.324: E/Unity(13663): 	at org.webrtc.PeerConnectionFactory.initialize(PeerConnectionFactory.java:217)
07-06 11:04:51.324: E/Unity(13663): 	at com.unity3d.player.UnityPlayer.nativeRender(Native Method)
07-06 11:04:51.324: E/Unity(13663): 	at com.unity3d.player.UnityPlayer.c(Unknown Source:0)
07-06 11:04:51.324: E/Unity(13663): 	at com.unity3d.player.UnityPlayer$e$2.queueIdle(Unknown Source:72)
07-06 11:04:51.324: E/Unity(13663): 	at android.os.MessageQueue.next(MessageQueue.java:405)
07-06 11:04:51.324: E/Unity(13663): 	at android.os.Looper.loop(Loo
</code>
</details>
<details open><summary>On Create Offer Click</summary>
<code>
07-06 11:05:23.016: E/Unity(13663): InvalidOperationException: Cannot create an offer with an uninitialized peer.
07-06 11:05:23.016: E/Unity(13663):   at Microsoft.MixedReality.WebRTC.Unity.PeerConnection.StartConnection () [0x00008] in <f1d52288d0e94ed0bd9bedda96f5797f>:0 
07-06 11:05:23.016: E/Unity(13663):   at NodeDssSignalerUI.<Start>b__5_0 () [0x0003d] in <17dc857eb7cf4642957e9a2d60ef8b54>:0 
07-06 11:05:23.016: E/Unity(13663):   at UnityEngine.Events.InvokableCall.Invoke () [0x00011] in <d71499eab75645c19df09ba17af9a506>:0 
07-06 11:05:23.016: E/Unity(13663):   at UnityEngine.Events.UnityEvent.Invoke () [0x00023] in <d71499eab75645c19df09ba17af9a506>:0 
07-06 11:05:23.016: E/Unity(13663):   at UnityEngine.UI.Button.Press () [0x00027] in <f1c52274b10c42898fd01b5f0aed5d19>:0 
07-06 11:05:23.016: E/Unity(13663):   at UnityEngine.UI.Button.OnPointerClick (UnityEngine.EventSystems.PointerEventData eventData) [0x00011] in <f1c52274b10c42898fd01b5f0aed5d19>:0 
07-06 11:05:23.016: E/Unity(13663):   at UnityEngine.EventSystems.ExecuteEvents.Execute (UnityEngine.EventSystems.IPointerClickHandler handler, UnityEngine.EventSystems.BaseEventData eventData) [0x00008] in <f1c52274b10c42898fd01b5f0aed5d19>:0 
07-06 11:05:23.016: E/Unity(13663):   at UnityEngine.EventSystems.ExecuteEvents.Execute[T] (UnityEngine.GameObject target, Uni</code>
</details>

**Screenshots**
_What I see on my phone_
![image](https://user-images.githubusercontent.com/16071761/86576859-a3e71580-bf79-11ea-97e5-f0a7aac87d72.png)

**Environment**
_Please fill the information for **each peer** if different_
 - Platform: Unity Android Player
- Phone: OnePlus 5T (A5010)
- Android Version: 10
 - Architecture: ARM64
 - Unity version (if applicable): 2018.4
 - Target device: Android Phone

**Additional context**
_I built the .aar using Android Studio on WSL2 (after copying this repo to my linux home because of some odd permission problems I was facing). When building, I faced the issue described in https://github.com/microsoft/MixedReality-WebRTC/issues/396, I encountered the error "no template named 'atomic_uint32_t' in namespace 'std'; did you mean 'atomic_init'?". I tried to heed @djee-ms 's warning and refrain from using a more recent NDK, but I could not figure out a way of resolving the issue otherwise. I ended up using NDK 18.x provided by Android Studio, which resolved the issue, and I hope(d) would not cause an issue. Furthermore, I was facing issues with sending cleartext data over http, which I resolved by adding android:usesCleartextTraffic="true" in the manifest. Additionally I added the internet, camera, and audio recording permissions to the android manifest as pointed out in https://github.com/microsoft/MixedReality-WebRTC/issues/329. Although, I still get no prompt in Android, and am unable to see myself._

**Comment by OmarAYousry** [comment]
I did a bit more digging around and resolved the issue. Turns out it was some issue where I wasn't actually building for x64 properly. I had to go into the player settings, enable IL2CPP as the scripting backend (instead of Mono), then I was able to enable building for ARM64 on Android. I seem to possibly be having a separate issue with permissions not (always?) popping up correctly. But I remember seeing another issue on the topic so hopefully I will sort it out.
